### DISCUSSION TITLE
Simulation of Multiple Receivers with Multiple Transmitters

### DISCUSSION THREAD

**Discussion by anindha-dhar** [NVlabs/sionna#385] [Q&A]
I want to simulate a cellular network. 

There will be **multiple receivers as well as multiple transmitters**. The users have _mobility_ (Considering Gauss Markov Mobility Model for it).

There will be a threshold BER value. If the current BER crosses the threshold_BER , then a handover will occur to the next nearest transmitter, and the transmission will resume, with the BER being measured again. Since the receivers are moving, if the BER again crosses the threshold, another handover will be performed, and the cycle will continue. 

**My question is how can I effectively simulate this scenario using sionna?** 

Like I will add a transmitter and a receiver into the scene. I will add mobility to the receiver and start computing paths. From the paths I can get '**a' & 'tau'** which can be used for calculating BER. Then if the BER crosses the threshold, I will remove the existing transmitter and add another nearest transmitter into the scene and then again start computing paths. This is the working methodology I have thought for single receiver and multiple transmitter. 

Is the procedure right or efficient? Alternatively, is there a more efficient method to simulate the scenario in sionna? How can I do this for multiple users with multiple transmitters? Do I have to remove user and add another user to the scene to compute paths? 

I'm new to sionna as well as in the field of wireless communications. I'm asking help from the experts. Any advice or suggestion will surely help me a lot.

**Comment by jhoydis** [comment]
Hi @anindha-dhar,

I think that what you want to do is some kind of system-level simulation with physical layer abstraction. That means that rather computing exact BER values by simulation, you would compute the SINR at a given position and then estimate the resulting BER from it.

In principle, you could compute coverage maps for all transmitters in the scene and then compute for each transmitter some kind of signal to interference map. Users would always seek to handover to another transmitter when the SIR is too low.
 
You would not even need to compute propagation paths with this approach and simply compute the coverage maps once. 

Hope this helps.

**Reply by anindha-dhar** [reply]
Thanks for the reply, it's really a great approach. But my question is how can I compute SINR at a given position and then estimate the resulting BER as you suggested:
> you would compute the SINR at a given position and then estimate the resulting BER from it.

**_Is there any pre-defined function available in sionna for computing SINR ?_**

Besides, could you please elaborate this process?:
>  compute for each transmitter some kind of signal to interference map. Users would always seek to handover to another transmitter when the SIR is too low.

 I can generate coverage map using sionna but I'm unsure about the process of Signal to Interference Mapping. Is it mentioned in the documentation? 

Thank you very much for the information. I'm looking forward to learn something more from you.

**Reply by anindha-dhar** [reply]
Hello,
From the documentation of **Coverage_Maps** , I have learned something called [**_sample_positions_**](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.CoverageMap.sample_positions) . I can set a min and max path gain as well as distance using this function. Can I use this for my desired scenario? 

I will sample positions with a defined max_path_gain & min_path_gain for a transmitter. If the user position doesn't belong to the same positions I got, then I will initiate a handover and switch to another transmitter. Is this procedure right? 

@jhoydis  @faycalaa any suggestion or advice will help me a lot. I'm stuck at this problem for a long time. 

Thanks.

**Comment by SebastianCa** [comment]
Hi @anindha-dhar,

Sionna does not have a pre-defined function to compute SINR directly via the API. Therefore, you’ll need to implement this by yourself. Here’s a basic approach to get you started:

1.	Compute SINR:
	- Signal Power (S): Calculate the received signal power from the desired transmitter at the receiver’s position.
	- Interference Power (I): Calculate the sum of the received power from all other interfering transmitters.
	- Noise Power (N): Define the noise power based on the system’s noise figure.
	
The SINR can be computed as:

$\text{SINR} = \frac{S}{I + N}$

2.	Estimate BER:
	- Once you have the SINR, you can estimate the BER using standard BER vs. SINR curves for the modulation scheme you are using. You would need to run Monte-Carlo simulations using your exact system model.


The `sample_position` utility function allows to randomly select user positions within a pre-defined range of parameters (such as the min/max path gain). Not sure if this helps for your handover problem, though it might be useful to draw random user positions for your Monte-Carlo simulations.

Hope this helps!

**Reply by lufeiaixuexi** [reply]
Hello, I would like to ask you a few points,
1.Is the tensor in cm obtained from the coverage map the Signal Power (S)?
2.How to obtain the corresponding Interference Power (I) in both single and multiple transmission points?
3.How to obtain Noise Power (N)?
Thank you very much！

**Comment by faycalaa** [comment]
Hi,

The next release of Sionna will include a feature for computing SINR maps, which should help address your issue.

**Reply by lufeiaixuexi** [reply]
Hello, may I ask when the new version of Sionna will be released?
